### DISCUSSION TITLE
How to hide a single node, when filter is active?

### DISCUSSION THREAD

**Discussion by theelectricsoul** [freeplane/freeplane#995] [Help]
Hello

My map is filtered but there is just one node that I do not want to see. How can I remove this node from the filter, without fiddeling with all the other nodes? I am looking for a one-click solution. 

thank you :)

**Comment by abc16361** [comment]
See menu View -> Hidden node.

**Reply by theelectricsoul** [reply]
I was not precise enough :) How can I remove one single node from the filter ? :)

**Reply by abc16361** [reply]
If you need to use Filter, then use Filter -> Compose filter to set up some conditions. For example, to filter out a node, you can add an attribute name like ' aaa' to the node, and use the Filter Composer, firstly to add an attribute condition 'Attribute name or value Contains "aaa", then select it and add a 'Not' condition by clicking the ' - ' button. When these conditions are done, you can use the filter dropdown box to filter out the node based on the condition. Similarly, you can do this also on node text, icons, hyperlinks etc.

![FilterOut1](https://user-images.githubusercontent.com/101362122/217672921-45a66516-f6fd-4e58-b9c3-22698af1a511.png)
![FilterOut2](https://user-images.githubusercontent.com/101362122/217672928-2af1d488-b68f-45e2-bb73-6d1a0e970314.png)

**Reply by theelectricsoul** [reply]
Sure...  but I think my requirement in the first post was missed ;) I do not want to fiddle around just to remove one node. I do not want my colleagues to wait for me clicking all over the place. 

In Freeplane you can ffilter fast by a selected node. Awesome. I think it should be possible to also remove one single node from view/filter.


Looks like a feature request :)   Thank you

**Comment by euu2021** [comment]
This script does the trick:

```
def showAncestors = true
def showDescendants = false
def nodeList = [ ]

c.find { it.isVisible() }.each { nodeList.add(it) }

nodeList -= node

c.select(node.parent)

map.filter(showAncestors, showDescendants, {it in nodeList})
```

But, it leaves you with the "showDescendants" off. So, then one idea is to have your usual "filter selected nodes" as a script that simply turns on the "showDescendants" before actually applying "filter selected nodes":

```
def showAncestors = true
def showDescendants = true
def nodeList = [ ]

nodeList = c.selecteds

map.filter(showAncestors, showDescendants, {it in nodeList})
```

**Reply by euu2021** [reply]
so, you use the second script to do your usual "filter selected nodes". Then, when you want to kick a node off of the party, you select it and apply the first script.

Side note: if you are interested in bringing connected nodes into the party,[ see this thread](https://sourceforge.net/p/freeplane/discussion/758437/thread/00a59ea0d6/).

**Comment by abc16361** [comment]
I just finished this script, working well on my test map. It works by toggling on/off selected nodes displaying.
[TogglingSelectedNodes.zip](https://github.com/freeplane/freeplane/files/10703339/TogglingSelectedNodes.zip)

```

//@ExecutionModes({ON_SINGLE_NODE})
//toggle on/off selected nodes displaying 
	

def targetNode = c.find{it.attributes.getNames().contains("#hidden")}
def selectedNodes = c.selecteds

if (targetNode == []) {
	selectedNodes.each { sl ->
		sl["#hidden"] = ""
	}
	def visibles = c.find{it.isVisible()}
	visibles -= selectedNodes
	node.map.filter { it in visibles }
	c.select(visibles[0])
}
else {
	targetNode.each { tg ->
		tg.attributes.removeAll("#hidden")
	}
	def visibles = c.find{it.isVisible()}
	visibles += targetNode
	node.map.filter { it in visibles }
	c.select(targetNode)
}

```
